html_text/html_text.py

```diff
@@ -132,17 +132,17 @@ def add_text(text_content, context):
         chunks.extend([space, text])
         context.prev = text_content
 
-    def traverse_text_fragments(tree, context, handle_tail=True):
-        """ Extract text from the ``tree``: fill ``chunks`` variable """
-        add_newlines(tree.tag, context)
-        add_text(tree.text, context)
-        for child in tree:
-            traverse_text_fragments(child, context)
-        add_newlines(tree.tag, context)
-        if handle_tail:
-            add_text(tree.tail, context)
-
-    traverse_text_fragments(tree, context=Context(), handle_tail=False)
+    # Extract text from the ``tree``: fill ``chunks`` variable
+    context = Context()
```

Comment:
It seems we can drop Context, and just use nonlocal. Not required to get this PR merged, of course :)

Reply:
Thanks @kmike implemented in https://github.com/zytedata/html-text/pull/10